src/postprocessing/least_squares.jl

```diff
@@ -226,27 +244,45 @@ function newtonls(res, w, x0, niters=5)
     npar = length(x0)
     # Mean square residual
     Q = chi2(res, w)/nobs
+    # Vector of x 
+    x = Matrix{T}(undef, niters + 1, npar) 
     # First guess
-    x_new = x0
+    x[1, :] = x0
+    # Vector of errors 
+    error = Vector{T}(undef, niters + 1)
+    # Error of first guess 
+    error[1] = T(Inf)
     # Iteration
     for i in 1:niters
-        # Gradient of Q with respect to x_0
-        dQ = TaylorSeries.gradient(Q)(x_new)
-        # Hessian of Q with respect to x_0
-        d2Q = TaylorSeries.hessian(Q, x_new)
+        # Gradient of Q with respect to x
+        dQ = TaylorSeries.gradient(Q)(x[i, :])
+        # Hessian of Q with respect to x
+        d2Q = TaylorSeries.hessian(Q, x[i, :])
         # Newton update rule
         Δx = - inv(d2Q)*dQ
-        x_new = x_new + Δx
+        # New x 
+        x[i+1, :] = x[i, :] + Δx
         # Normal matrix
         C = d2Q/(2/nobs) # C = d2Q/(2/m)
-        @show sqrt(((Δx')*(C*Δx))/npar)
+        # Error 
+        error2 = ( (Δx') * (C*Δx) ) / npar
+        if error2 ≥ 0
+            error[i+1] = sqrt(error2)
+        # The method do not converge
+        else 
+            return false, x[i+1, :], inv(C)
+        end 
     end
+    # Index with the lowest error 
+    i = argmin(error)
```

Comment:
Note that `findmin` returns the index and the value of the minimum.

Reply:
Both `diffcorr` and `newtonls` only need the index, so I think it is better to use only `argmin`.

Reply:
If the index is only needed, I agree; let's leave it as it is now.